core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -245,9 +269,21 @@ public boolean sync() {
     if (LOG.isDebugEnabled()) {
       start = System.currentTimeMillis();
     }
-
-    try {
-      syncInodeMetadata(mRootPath);
+    try (LockedInodePath path = mInodeTree.lockInodePath(mRootScheme)) {
```

Comment:
This will relock and recheck permissions? Can we add that to the comments of this method or class?

Reply:
> This will relock and recheck permissions

The permission check and locking were all moved into this block. The `syncMetadata` function in DFSM no longer does the lock and permission check